SOLR-16962: Restore ability to configure tlog directory

magibney opened the pull request:
backported from in-process upsteam PR at commit ea8e829d78d207b4a6e48f4514737bf6f717c7d4

also includes revert of backport of earlier iteration of same upstream PR

magibney committed:
Revert "SOLR-16962: Restore ability to configure tlog directory (#132)"

This reverts commit f4cee667c17fcb5ed17440255786aa93cba8bd01.

magibney committed:
SOLR-16962: Restore ability to configure tlog directory

backported from in-process upsteam PR at commit ea8e829d78d207b4a6e48f4514737bf6f717c7d4

magibney committed:
fix missing import

hiteshk25 commented on solr/core/src/test-files/solr/configsets/cloud-hdfs/conf/solrconfig.xml:
```diff
@@ -35,7 +35,9 @@
       <softCommit>${solr.commitwithin.softcommit:true}</softCommit>
     </commitWithin>
-    <updateLog></updateLog>
+    <updateLog>
```
@magibney this satisfies our usecase test?

magibney replied:
This (hdfs) is actually irrelevant to our use case. It's even not directly relevant to the PR (which doesn't really exercise explicit `solr.ulog.dir` configuration for hdfs), but it puts a hook in the config file to make such testing possible. This PR _does_ test regular local filesystem explicit `solr.ulog.dir`, which is our usecase (and the most common usecase generally, by far).

hiteshk25 reviewed (approved):
LGTM (Thanks @magibney. Please see if you can our usecase explicitly )

magibney committed:
remove redundant arg

backported from in-process upsteam PR at commit 794753e61e4a0b7f690938c8579bf57cfd460ced

magibney commented:
merging this to `branch_9_3`; upstream PR is almost complete.
